kong/plugins/aws-lambda/iam-ec2-credentials.lua

```diff
@@ -9,29 +9,71 @@ local kong = kong
 local METADATA_SERVICE_PORT = 80
 local METADATA_SERVICE_REQUEST_TIMEOUT = 5000
 local METADATA_SERVICE_HOST = "169.254.169.254"
-local METADATA_SERVICE_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
+local METADATA_SERVICE_TOKEN_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
+                             "/latest/api/token"
+local METADATA_SERVICE_IAM_URI = "http://" .. METADATA_SERVICE_HOST .. ":" .. METADATA_SERVICE_PORT ..
                              "/latest/meta-data/iam/security-credentials/"
 
 
-local function fetch_ec2_credentials()
+local function fetch_ec2_credentials(config)
   local client = http.new()
   client:set_timeout(METADATA_SERVICE_REQUEST_TIMEOUT)
-  local role_name_request_res, err = client:request_uri(METADATA_SERVICE_URI)
+
+  local protocol_version = config.aws_imds_protocol_version
+  local imds_session_headers
+
+  if protocol_version == "v1" then
+
```

Comment:
```suggestion
```

Reply:
Sorry, I overlooked this comment - I opened a new PR to get rid of the style issue and will create a matching docs update.

If you could approve https://github.com/Kong/kong/pull/9985, it'd be appreciated.